### PR TITLE
Auto-initialize DPE with the RT Journey PCR

### LINKED ISSUE #1118: [RT Test] Check that measurements are stored in DPE when STASH_MEASUREMENT is called

### LINKED ISSUE #1169: Rethink tagging behavior
Few questions that need to be answered regarding tagging:

How should tagging work for retired contexts? I see 3 options:

1. Disallow TAG_TCI and GET_TAGGED_TCI on retired contexts
2. Allow TAG_TCI and GET_TAGGED_TCI on retired contexts
3. Disallow TAG_TCI but allow GET_TAGGED_TCI on retired contexts

We currently do 1, but I think 3 could be better. This would mean users would be able to tag a context, derive a child from it, and still have access to its TCIs directly. 

If we tag a default context, and then derive a child from that context with retains-parent = false, and make-default = true, we would get a new context with the same handle as the tagged context, except without the tag. Should the tag carry over to the new context?

### PR DESCRIPTION
**Pull request by sree-revoori1**
Currently, there is a bug in RT where Drivers::get_dpe_root_context_idx returns the incorrect DPE context index. This function searches for the context whose parent is 0xFF, which indicates that that context is the root. However, DPE is auto-initialized and then DeriveChild is called in order to measure the RT journey PCR into a child context. This means the root context does not contain the RT journey PCR which is incorrect.

With the change in https://github.com/chipsalliance/caliptra-dpe/pull/278, we are able to auto-initialize DPE with a measurement. Using that, we now auto-initialize DPE with the RT journey PCR so that we don't have an extraneous default measurement in DPE. 

This PR also updates tagging behavior to allow GET_TAGGED_TCI on retired contexts and disallow TAG_TCI on retired contexts. The test has been updated accordingly.

This PR also adds a test that checks that measurements are stored in DPE when STASH_MEASUREMENT is called.

Fixes #1118 
Fixes #1169

### CODE REVIEW

File: runtime/src/invoke_dpe.rs [hunk 1]
```diff
@@ -131,4 +128,21 @@ impl InvokeDpeCmd {
...
+    pub fn clear_tags_for_inactive_contexts(
+        dpe: &mut DpeInstance,
+        context_has_tag: &mut [U8Bool; MAX_HANDLES],
+        context_tags: &mut [u32; MAX_HANDLES],
+    ) {
+        (0..MAX_HANDLES).for_each(|i| {
+            if i < dpe.contexts.len()
+                && i < context_has_tag.len()
+                && i < context_tags.len()
+                && dpe.contexts[i].state != ContextState::Active
```
Comment (jhand2) [inline runtime/src/invoke_dpe.rs:141]: Should this be for `dpe.contexts[i].state == ContextState::Inactive`?

Comment (sree-revoori1) [reply runtime/src/invoke_dpe.rs]: Yes, thanks for catching. Done.

File: dpe
Comment (jhand2) [inline dpe:1]: I think this DPE commit doesn't exist in caliptra-dpe upstream

Comment (sree-revoori1) [reply dpe:1]: Fixed

**Review by jhand2 — APPROVED** [review]

**Pull request merged**